plugins/pulp_rpm/yum_plugin/util.py

```diff
@@ -173,3 +175,40 @@ def generate_listing_files(root_publish_dir, repo_publish_dir):
 
         # work up the directory structure
         working_dir = os.path.dirname(working_dir)
+
+
+def errata_format_to_datetime(datetime_str, msg):
+    """
+    Convert known errata date-time formats to datetime object.
+
+    Expected formats are:
+     - '%Y-%m-%d %H:%M:%S'
+     - '%Y-%m-%d %H:%M:%S UTC'
+
+    :param datetime_str: date and time in errata specific format
+    :type  datetime_str: str
+
+    :param msg: additional error message in case of exception
+    :type  msg: str
+
+    :return: parsed date and time
+    :rtype: datetime.datetime
+    :raises ValueError: if the date and time are in unknown format
+    """
+    strptime_pattern = '%Y-%m-%d %H:%M:%S'
+    match_and_drop_utc_regexp = r'^\s*(?P<datetime>.*?)( UTC)?\s*$'
+    err_msg = 'Unknown format: unable to convert "%s" to the datetime object' % datetime_str
+
+    # we do not want timezone other than UTC, if it is present
+    match_obj = re.match(match_and_drop_utc_regexp, datetime_str)
+    if match_obj is None:
+        raise ValueError(err_msg, msg)
+    datetime_str = match_obj.group('datetime')
+
+    try:
+        datetime_obj = datetime.datetime.strptime(datetime_str, strptime_pattern)
+    except ValueError as e:
+        e.args += (err_msg, msg)
+        raise
```

Comment:
I've never seen modification to the last error and then the raise. It makes sense though. I learned something. :palm_tree: :ocean: :palm_tree: